public/javascripts/EditCartItems.js

```diff
@@ -0,0 +1,290 @@
+                        event.data.row.dirty = true;
+                        editCartItems.populateList();
+                    })
+                    .appendTo(editRow.optionsRow);
+
+                $(document.createElement("span"))
+                    .text(" = ")
+                    .addClass("float_left operator")
+                    .appendTo(row.optionsRow);
+                $(document.createElement("span"))
+                    .text(" = ")
+                    .addClass("float_left operator")
+                    .appendTo(editRow.optionsRow);
+
+                row.rowTotal = $(document.createElement("span"))
+                    .text(product.rows[i].currentSize.Size * product.rows[i].quantity)
+                    .addClass("float_left row_total")
+                    .appendTo(row.optionsRow);
+                editRow.rowTotal = $(document.createElement("span"))
+                    .text(product.rows[i].currentSize.Size * product.rows[i].quantity)
+                    .addClass("float_left row_total")
+                    .appendTo(editRow.optionsRow);
+
+                row.color = $(document.createElement("span"))
+                    .text("Run Color: " + product.rows[i].color)
+                    .addClass("float_right run_color")
+                    .appendTo(row.optionsRow);
+                editRow.color = $(document.createElement("span"))
+                    .text("Run Color: " + product.rows[i].color)
+                    .addClass("float_right run_color unimplemented")
+                    .appendTo(editRow.optionsRow);
+
+
+                row.location = $(document.createElement("span"))
+                    .text("Location: " + product.rows[i].location)
+                    .addClass("float_right location")
+                    .appendTo(row.optionsRow);
+                editRow.location = $(document.createElement("span"))
+                    .text("Location: " + product.rows[i].location)
+                    .addClass("float_right location unimplemented")
+                    .appendTo(editRow.optionsRow);
+                $(document.createElement("br")).appendTo(row.optionsRow);
+
+                if(product.editing) {
+                    row.optionsRow.hide();
+                    editRow.optionsRow.show();
+                } else {
+                    row.optionsRow.show();
+                    editRow.optionsRow.hide();
+                }
+                cart.itemRows.push(row);
+                cart.editRows.push(editRow);
+            }
+            itemList.append(cart.cartItem);
+        });
+        inventory_container.append(itemList);
+    },
+
+    editItem: function(index) {
+
+        var product = editCartItems.model.products[index];
+        product.editing = !product.editing;
+
+        if(!product.editing) {
+            editCartItems.submitDirtyItems(product);
+        }
+        editCartItems.populateList();
+    },
+
+    submitDirtyItems: function(product) {
```

Comment:
Loving this way of doing the writeback. There is a convention btw in Javascript for things like the dirty bit, as a private member that should not be used from outside the class, typically JS developers prepend an underscore. It does not make the variable private, but it does tell other developers it is not for them to utalize. So I might do product.rows[i]._dirty.